core/src/main/java/org/infinispan/AdvancedCache.java

```diff
@@ -426,6 +430,64 @@
    void putForExternalRead(K key, V value, Metadata metadata);
 
    /**
+    * An overloaded form of {@link #compute(K, BiFunction)}, which takes in an
+    * instance of {@link Metadata} which can be used to provide metadata
+    * information for the entry being stored, such as lifespan, version
+    * of value...etc.
+    *
+    * @param key key with which the specified value is associated
+    * @param remappingFunction function to be applied to the specified key/value
+    * @param metadata information to store alongside the new value
+    * @return the previous value associated with the specified key, or
+    *         <tt>null</tt> if remapping function is gives null.
+    *
+    * @since 9.x
+    */
+   V compute(K key, BiFunction<? super K, ? super V, ? extends V> remappingFunction, Metadata metadata);
+
+   /**
+    * An overloaded form of {@link #computeIfPresent(K, BiFunction)}, which takes in an
+    * instance of {@link Metadata} which can be used to provide metadata
+    * information for the entry being stored, such as lifespan, version
+    * of value...etc. The {@link Metadata} is only stored if the call is
+    * successful.
+    *
+    * @param key key with which the specified value is associated
+    * @param remappingFunction function to be applied to the specified key/value
+    * @param metadata information to store alongside the new value
+    * @return the previous value associated with the specified key, or
+    *         <tt>null</tt> if there was no mapping for the key.
+    *
+    * @since 9.x
+    */
+   V computeIfPresent(K key, BiFunction<? super K, ? super V, ? extends V> remappingFunction, Metadata metadata);
+
+   default V computeIfPresent(K key, SerializableBiFunction<? super K, ? super V, ? extends V> remappingFunction, Metadata metadata) {
```

Comment:
Add a comment which at least references the other computeIfPresent method so people know to look there for the real description.